libs/vkd3d/command.c

```diff
@@ -9559,28 +9593,25 @@ static void STDMETHODCALLTYPE d3d12_command_list_DiscardResource(d3d12_command_l
     all_subresource_full_discard = first_subresource == 0 && subresource_count == resource_subresource_count;
     d3d12_command_list_track_resource_usage(list, texture, !all_subresource_full_discard);
 
-    if (all_subresource_full_discard)
-    {
-        has_bound_subresource = false;
-        has_unbound_subresource = false;
+    has_bound_subresource = false;
+    has_unbound_subresource = false;
 
-        /* If we're discarding all subresources, we can only safely discard with one barrier
-         * if is_bound state is the same for all subresources. */
-        for (i = first_subresource; i < first_subresource + subresource_count; i++)
-        {
-            vk_subresource = d3d12_resource_get_vk_subresource(texture, i, false);
-            vk_subresource_layers = vk_subresource_layers_from_subresource(&vk_subresource);
-            attachment_idx = d3d12_command_list_find_attachment_view(list, texture, &vk_subresource_layers);
+    /* If we're discarding all subresources, we can only safely discard with one barrier
+      * if is_bound state is the same for all subresources. */
+    for (i = first_subresource; i < first_subresource + subresource_count; i++)
+    {
+        vk_subresource = d3d12_resource_get_vk_subresource(texture, i, false);
 
-            is_bound = attachment_idx >= 0 && (list->rendering_info.state_flags & (VKD3D_RENDERING_ACTIVE | VKD3D_RENDERING_SUSPENDED));
-            if (is_bound)
-                has_bound_subresource = true;
-            else
-                has_unbound_subresource = true;
-        }
+        if (d3d12_command_list_is_subresource_bound_as_rtv_dsv(list, texture, &vk_subresource))
+            has_bound_subresource = true;
+        else
+            has_unbound_subresource = true;
+    }
 
+    if (all_subresource_full_discard)
```

Comment:
It took a good minute to understand what this branch was supposed to do. It would help to bind this logic to something like `bool discard_partially_overlaps_render_targets` (or something less jank) I think